openmetadata-ui/src/main/resources/ui/src/pages/teams/AddUsersModalV1.tsx

```diff
@@ -140,70 +136,75 @@ const AddUsersModalV1 = ({
   };
 
   const handleSearchAction = (searchValue: string) => {
+    setUniqueUser([]);
+    setCurrentPage(INITIAL_FROM);
     setSearchText(searchValue);
+    if (searchValue) {
+      searchUsers(searchValue, currentPage);
+    } else {
+      fetchAllUsers();
+    }
   };
-
-  useEffect(() => {
-    if (allUsers.length > 0) {
-      getUniqueUserList();
+  const onScroll = (e: React.UIEvent<HTMLElement, UIEvent>) => {
+    if (
+      e.currentTarget.scrollHeight - e.currentTarget.scrollTop ===
+      ContainerHeight
+    ) {
+      if (searchText) {
+        // make API call only when current page size is less then total count
+        PAGE_SIZE_MEDIUM * currentPage < totalESCount &&
+          searchUsers(searchText, currentPage);
```

Comment:
onScroll it shouldn't be searchUsers

Reply:
onScroll if there is a search term present, we need to fetch data from Search API, but if there is not search term then we are fetching from REST API